tests/testsuite/alt_registry.rs

```diff
@@ -307,46 +312,26 @@ fn publish_with_registry_dependency() {
 
     Package::new("bar", "0.0.1").alternative(true).publish();
 
-    // Login so that we have the token available
-    p.cargo("login --registry alternative TOKEN").run();
-
-    p.cargo("publish --registry alternative").run();
-
-    validate_alt_upload(
-        r#"{
```

Comment:
Did the verification of the JSON payload for `publish` get removed in this change? I think we want to keep that.

Reply:
I think it is OK to remove it. Usually, we have two reasons for the verification of the JSON.
1. We want to make sure we access the right API.
2. We want to make sure we store the right content.

But for this case, we use stderr to make sure we access the alternative registry. And also this is nothing special in this crate file, so I think we can safely remove it. I only keep the verification if we have some special content in the publish request.

Reply:
This is doing more than just validating that it goes to the correct registry. For example, it is validating that the `deps` field also has the correct registry value (in this case, unset). I think it would be good to avoid removing test coverage.